### DISCUSSION TITLE
3d+ (4d+ and 3d) stuff maybe

### DISCUSSION THREAD

**Discussion by maybeJosiah** [community/community#142160] [Programming Help]
### Body

Hello and thanks for this community, I am Josiah, I do 3d+ stuff and timestamps stuff, some 3d+ stuff is open source, StinomXE online is like mine, I am not much on GitHub because I already pay minimally for web hosting and GitHub seems like just another expense, I even locally version control maybe or maybe not, maybe. Anyone want to talk math and 2d like polygon cutters and other 2d stuff with me? I know lots of programming stuff and am currently looking for some reviewing of code and other stuff (math stated), well, said all I want now, if this is bad or something can you feel free to delete moderators (or anyone knowing enough)? Anyone else have any thoughts on 3 or more dimensional math and programming, novelty though it is? Sorry if I promote, I kind of don't know how else to ask maybe or maybe not, maybe.

Edit in:
I know I can write entire screens full of algorithms before actually implementing so if you want them I am okay with sharing but just warning, it will take a long ramble maybe or maybe not, maybe. Currently StinomXE runs on averages, cutters of interest are 2d polygon in 2d polygon part or whole, infinite line on sides, between 2 infinite lines creating a trapezoid, 2d line in polygon, and others, all reducing to 2d basically maybe or maybe not, maybe.

Second edit: I think I stated all like my algorithms now, reviews?
Third edit: To be clear, looking for code reviews of StinomXE and algorithms as stated below maybe or maybe not, maybe.

### Guidelines

- [x] I have read and understood this category's [guidelines](https://github.com/orgs/community/discussions/51384) before making this post.

**Comment by ebndev** [comment]
Hi @maybeJosiah, thanks for being a part of the GitHub Community!

I've removed a link from your post as we do not allow personal identifying information to be shared in Discussions.

Hope to see you around in other discussions in the future ⭐

**Reply by maybeJosiah** [reply]
Okay, thanks for telling me maybe or maybe not, maybe.

**Reply by maybeJosiah** [reply]
I redacted other links in another forum too, anyway, no links to my site maybe or maybe not, maybe.

**Comment by luannndev** [comment]
Hi Josiah,

Welcome to the community! It's great to see someone passionate about 3D and mathematical programming. I’d love to hear more about your open-source projects and the specific areas of 3D+ stuff you’re working on.

Polygon cutting in 2D is a fascinating topic! Are you exploring any particular algorithms or applications? It might also be interesting to discuss how those concepts could be applied to 3D geometry.

If you’re looking for code reviews, feel free to share snippets or specific areas where you’d like feedback. The community is a fantastic place for collaboration and learning, so don’t hesitate to reach out!

Looking forward to your reply!

**Reply by maybeJosiah** [reply]
Okay, I guess since like my repo is off GitHub now you can internet look up StinomXE, mods feel free to correct if I cannot even share that, it is actually a thing like 4d, 5d, etc. dimensional and also 3d maybe or maybe not, maybe. I am looking for feedback on code (StinomXE, Ruby, JS) and math stuff currently, basically what it does is all for draw order and like thing A is partially in front or behind thing B or fully in front or behind and how to tell programmatically maybe or maybe not, maybe.

I know I can write entire screens full of algorithms before actually implementing so if you want them I am okay with sharing but just warning, it will take a long ramble maybe or maybe not, maybe. Currently StinomXE runs on averages, cutters of interest are 2d polygon in 2d polygon part or whole, infinite line on sides, between 2 infinite lines creating a trapezoid,  2d line in polygon, and others maybe or maybe not, maybe. Anyway, just ask if you want an algorithm or whatever to review maybe or maybe not, maybe.

**Comment by Annishivakumara** [comment]
Hello

**Comment by senulahesara** [comment]
Hi Josiah! Welcome to the community! It’s great to hear about your work in 3D and timestamps, and it's always awesome when people share their open-source contributions. Regarding GitHub, while it may seem like an additional expense, there are free tiers that allow you to host and version control projects. It can be a useful way to collaborate and showcase your work.

As for math and programming around polygon cutters and multi-dimensional work, that sounds super interesting! I’d love to discuss it further. Do you have any projects or specific areas in 2D/3D math you’re currently focused on? I'd be happy to check out your code or share insights!

Feel free to keep the conversation going, and don’t hesitate to reach out for code reviews or math discussions!

**Reply by maybeJosiah** [reply]
See above, I commented what classes of cutters and other things I am working on, and like my repo is off GitHub because I only get 500MB for free tier and I would like to stay okay for classes I am currently in, I have not written out cutter math algorithms yet but I can, you want me to write out one and which? It is Euclidean 4d+ and working from locked to dimension axes to any view point I am working on maybe or maybe not, maybe.

**Reply by maybeJosiah** [reply]
Does commenting here count toward that 500MB maximum free tier?

**Comment by maybeJosiah** [comment]
Okay, I will just start talking, so I am working with closed, 2d, straight line, not self intersecting polygons, lines, and points maybe or maybe not, maybe. To start, find signed area, sign meaning clockwise or counter clockwise. When cutting, we try for a complete shape so for points, return to start is desired and so is all included that should be maybe or maybe not, maybe. We start with on sides of an infinite line, get all intersects of line and polygon, find which is on a side and stay on that side until done on that side and repeat for other side maybe or maybe not, maybe. Count lines on infinite line as two end points, if not included in anything else, include them as one shape maybe or maybe not, maybe. 

Never start with an already used point and two lines ending at infinite line and both in way to include counts as 2 points maybe or maybe not, maybe. For which way to go, find a current to next, more in way as measured with signed area but less than to current to previous is in maybe or maybe not, maybe. I am counting a 4 way intersection (lines end at other line but don't pass) as self intersecting and on border as in and all out is out maybe or maybe not, maybe. We can always find which way we are considering "in" by finding it on plane and from a start to an end, which rotation way? Include infinite line and connected parts in shape taking note what end connects to what maybe or maybe not, maybe. That is all details of one cutter I think, am I missing anything? Can you tell me which cutters I said you want (below luannndev)?

**Comment by maybeJosiah** [comment]
For point in polgon, find direction of rotation, signed area, find all intercepts with a dimension line with a coordinate of point to consider, if no two sides and point between on that, not inside, but if two sides of point on that line then continue, find range of possible values of in for 2 closest points to point to be in or out, if not in an acceptable angle from either intercept then out, otherwise in maybe or maybe not, maybe. See [Other part](https://github.com/orgs/community/discussions/142160#discussioncomment-11008092) for what can be in on a polygon and which polygons this applies to maybe or maybe not, maybe. Asking for review, is this correct or missing anything and will it be 100% accurate?

**Comment by maybeJosiah** [comment]
[Other post](https://github.com/orgs/community/discussions/142160#discussioncomment-11008092), For finite line in 2d polygon, find what intersects infinitely stretching line of that finite line, find nearest outside finite part if there are any and from that find angles that are in and signed area and going same way as measured for signed area, find areas that are in polygon by possible angles maybe or maybe not, maybe.

**Comment by maybeJosiah** [comment]
Last but not easiest, 2d polygon to 2d polygon like before, [Other post](https://github.com/orgs/community/discussions/142160#discussioncomment-11008092), start with signed area of both, find all points that they intersect each other with that both in condition for counting twice a point, then starting from a point on one, go around other inside it by angles until reaching a connected to origin, use another outside to current shapes found point, continue until all found, if a line is exactly on another, count it like 2 points and if it could connect to itself, reserve for later when all but it is found as in maybe or maybe not, maybe. Also, things with no intersections, if one point of border of either is inside border of other then entire shape is in shape that point was in border of and if neither shape has one point inside like that, all outside maybe or maybe not, maybe. Here are all like my algorithms, opinions?

**Comment by ghostinhershell** [comment]
👋 Welcome to the GitHub Community on Discussions, @maybeJosiah Happy to have you here 🙂

To help keep our space clutter-free, we recommend introducing yourself in our [official introduction thread](https://github.com/orgs/community/discussions/23204) and checking out this Discussion to [Meet Your GitHub Community Team](https://github.com/orgs/community/discussions/28776).
